packages/plugin-workspace-tools/sources/commands/foreach.ts

```diff
@@ -5,12 +5,21 @@ import {formatUtils, miscUtils, structUtils}                         from '@yarn
 import {gitUtils}                                                    from '@yarnpkg/plugin-git';
 import {Command, Option, Usage, UsageError}                          from 'clipanion';
 import micromatch                                                    from 'micromatch';
-import {cpus}                                                        from 'os';
+import os                                                            from 'os';
 import pLimit                                                        from 'p-limit';
 import {Writable}                                                    from 'stream';
 import {WriteStream}                                                 from 'tty';
 import * as t                                                        from 'typanion';
 
+function availableParallelism() {
+  // TODO: Use os.availableParallelism directly when dropping support for Node.js < 19.4.0
+  if (`availableParallelism` in os)
+    // @ts-expect-error - No types yet
+    return os.availableParallelism();
+
+  return Math.max(1, os.cpus().length);
+}
```

Comment:
Is there a reason for duplicating this helper instead of extracting it to `nodeUtils`?

Reply:
No, fixed 👍